Automattic/jetpack
# Newsletter categories: Remove newsletter categories from subscription block

## Proposed changes:
This is a cleanup PR. Since we decided to move the newsletter categories to the subscribe modal, they're not needed in the subscribe block anymore.

This PR removes all trace of newsletter categories from the subscribe block. 

It does keep the endpoint (`WPCOM_REST_API_V2_Endpoint_Newsletter_Categories_List`) to retrieve the newsletter categories around, since we'll need that in order to show what categories the newsletter will be sent to in the pre-publush panel.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
1. Apply this PR to your JT-site
2. Add the `newsletter-categories` blog sticker to your JT-site & add some newsletter categories through Calypso
3. Create a new post/page with a subscribe block
4. View the page

Make sure that: 
- there are no errors
- the newsletter categories aren't displayed above the block
